content/blog/2025/per-user-storage-quota/index.md

```diff
@@ -0,0 +1,46 @@
+---
+title: "Enforcing per-user storage quotas with `jupyterhub-home-nfs`"
+subtitle: ""
+summary: ""
+authors: ["Sarah Gibson"]
+tags: []
+categories: []
+date: 2025-01-28T09:57:28+00:00
+lastmod: 2025-01-28T10:10:14+00:00
+featured: false
+draft: false
+
+# Featured image
+# To use, add an image named `featured.jpg/png` to your page's folder.
+# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
+image:
+  caption: ""
+  focal_point: ""
+  preview_only: false
+
+# Projects (optional).
+#   Associate this post with one or more of your projects.
+#   Simply enter your project's folder or file name without extension.
+#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
+#   Otherwise, set `projects = []`.
+projects: ["nasa-veda"]
+---
+
+When sharing a storage disk between users, as is usually the case in a JupyterHub deployment, it is important to put in guardrails so that one user cannot eat up the whole storage capacity from the rest of the users.
+To this end, 2i2c in close collaboration with [Development Seed](https://developmentseed.org) have developed the [`jupyterhub-home-nfs` project](https://github.com/2i2c-org/jupyterhub-home-nfs) which is a Helm chart that permits enforcing per-user quotas on the storage space.
+
+{{% callout note %}}
+Note that this feature is currently available to AWS hosted hubs only and will be rolled out to other cloud providers in the future.
+{{% /callout %}}
+
+Under the hood, the Helm chart runs [NFS Ganesha](https://github.com/nfs-ganesha/nfs-ganesha) as an in-cluster NFS server, backed by XFS as the underlying filesystem. Storage quota is enforced through XFS's native quota management utility `xfs_quota`.
+
+Since this feature moves our infrastructure away from managed filesystems (such as AWS's Elastic File System), we have also developed monitoring and alerting mechanisms that will let us know when the disks are getting full, and automated back-ups for disaster recovery.
```

Comment:
```suggestion
Since this feature moves our infrastructure away from managed filesystems (such as AWS's Elastic File System) that cannot support per-user storage quotas, we have also developed monitoring and alerting mechanisms that will let us know when the disks are getting full, and automated back-ups for disaster recovery.
```